opencart 1.5.6.4/upload/admin/model/smailyforopencart/admin.php

```diff
@@ -13,5 +13,21 @@ public function getSettingValue($group, $key, $store_id = 0) {
         } else {
             return null;    
         }
+    }
+    public function editSettingValue($group = '', $key = '', $value = '', $store_id = 0) {
+        // Use UPDATE if group exist, otherwise INSERT it into db.
+        if (in_array(0, $this->db->query("SELECT EXISTS(SELECT * FROM " . DB_PREFIX . "setting WHERE `key`='" . $this->db->escape($key) . "')")->row)) {
```

Comment:
I see a few issues with this implementation:
a) you are not using `$group` variable value to check if setting exists;
b) using `in_array` is too complex - you can simplify (remove `SELECT EXISTS`) query and use `$this->db->query(...)->num_rows` instead;